background/transition.js

```diff
@@ -1,19 +1,23 @@
-chrome.runtime.onInstalled.addListener(async (details) => {
-  const currentVersion = chrome.runtime.getManifest().version;
-  const [major, minor, patch] = currentVersion.split(".");
-  if (details.previousVersion && details.previousVersion.startsWith("0")) {
-    chrome.tabs.create({ url: "https://scratchaddons.com/scratch-messaging-transition" });
-  } else if (
-    details.reason === "install" &&
-    chrome.runtime.getManifest().version_name.includes("-prerelease") === false
-  ) {
-    chrome.tabs.create({ url: "https://scratchaddons.com/welcome" });
-  }
+chrome.storage.sync.get(["developerMode"], (r) => {
+  const developerMode =
+    r.developerMode === "always" ||
+    (r.developerMode === undefined && chrome.runtime.getManifest().version_name.endsWith("-prerelease"));
+  chrome.runtime.onInstalled.addListener(async (details) => {
+    const currentVersion = chrome.runtime.getManifest().version;
+    const [major, minor, patch] = currentVersion.split(".");
+    if (details.previousVersion && details.previousVersion.startsWith("0")) {
+      chrome.tabs.create({ url: "https://scratchaddons.com/scratch-messaging-transition" });
+    } else if (details.reason === "install" && !developerMode) {
+      chrome.tabs.create({ url: "https://scratchaddons.com/welcome" });
+    }
 
-  if (details.reason === "install") {
-    chrome.storage.local.set({
-      bannerSettings: { lastShown: `${major}.${minor}` },
-    });
+    if (details.reason === "install") {
+      chrome.storage.local.set({
+        bannerSettings: { lastShown: `${major}.${minor}` },
+      });
+    }
+  });
+  if (!isDevVersion) {
+    chrome.runtime.setUninstallURL("https://scratchaddons.com/farewell");
   }
```

Comment:
My current version when running this is 1.6.0-prerelease (so devMode is already enabled). When I turned off dev mode in the settings, there wasn't an un install link. So you need a listener for new more settings enabled to check if that setting was changed. But it also goes the opposite way too. If I were to install this extension as version 1.5.0 and in the settings turn on devMode setting, when I uninstall, I would still get an uninstall link.

Reply:
But (with my new saved settings) if I were to uninstall and reinstall again, then this wouldn't be a problem since it only decides these things when the extension is installed.